openapi_python_client/resolver/resolved_schema.py

```diff
@@ -0,0 +1,202 @@
+                else:
+                    refs_to_replace.append((owner, remote_schema, remote_value))
+
+        for owner, remote_schema, remote_value in refs_to_replace:
+            self._process_remote_components(remote_schema, remote_value, 1)
+            self._replace_reference_with(owner, remote_value)
+
+    def _process_remote_components(
+        self, owner: SchemaData, subpart: Union[SchemaData, None] = None, depth: int = 0
+    ) -> None:
+        target = subpart if subpart else owner
+
+        for parent, ref_key, ref_val in self._lookup_schema_references(target):
+            ref = Reference(ref_val)
+
+            if ref.is_local():
+                # print('Found local reference >> {0}'.format(ref.value))
+                if depth > 0:
+                    self._transform_to_local_components(owner, ref)
+            else:
+                remote_path = ref.pointer.value
+                if remote_path not in self._refs:
+                    self._errors.append("Failed to resolve remote reference > {0}".format(remote_path))
+                else:
+                    remote_owner = self._refs[remote_path]
+                    self._transform_to_local_components(remote_owner, ref)
+                    self._transform_to_local_ref(parent, ref)
+
+    def _transform_to_local_components(self, owner: SchemaData, ref: Reference) -> None:
+        self._ensure_components_dir_exists(ref)
+
+        # print('Processing remote component > {0}'.format(ref.value))
+        remote_component = self._lookup_dict(owner, ref.path)
+        pointer_parent = ref.pointer.parent
+
+        if pointer_parent is not None:
+            root_components_dir = self._lookup_dict(self._resolved_remotes_components, pointer_parent.value)
+            component_name = ref.path.split("/")[-1]
+
+        if component_name == "SorTransparentContainer" or component_name == "sorTransparentContainer":
+            print(ref.value)
+
+        if remote_component is None:
+            print("Weirdy relookup of >> {0}".format(ref.value))
+            assert ref.is_local() and self._lookup_dict(self._resolved_remotes_components, ref.path)
+            return
+
+        if "$ref" in remote_component:
+            subref = Reference(remote_component["$ref"])
+            if not subref.is_local():
+                print("Lookup remote ref >>> {0}".format(subref.value))
+                return self._process_remote_components(remote_component)
+
+        if root_components_dir:
+            if component_name in root_components_dir:
+                local_component_hash = self._reference_schema_hash(root_components_dir[component_name])
+                remote_component_hash = self._reference_schema_hash(remote_component)
+
+                if local_component_hash == remote_component_hash:
+                    return
+                else:
+                    pass
+                # print('=' * 120)
```

Comment:
Here we have a collision to handle, we have two remote components with the same name but different content, may be implemented later, but definitly required

Reply:
Also, we should probably extract the code responsible to detect collisions into its own class, thus later on we could customise collision detection with DI.
Assigned new reference name should also be handle there (Same goal, for easy behaviour customization)